academic-services/knowledge-exploration-service/reference-makes-command-line-tool.md

```diff
@@ -171,37 +196,287 @@ The input entities file Url prefix.
+
+```cmd
+kesm DescribeGrammar --CompiledGrammarFilePath
+```
+
+### DescribeGrammar Required Parameters
+
+`--CompiledGrammarFilePath`
+
+The file path to the compiled grammar binary. E.g. grammar.kesg
+
+## Interpret Command
+
+Interprets a natural language query using specified indexes and grammar.
+
+```cmd
+kesm Interpret --Query
+               --IndexFilePaths
+               --GrammarFilePath
+               [--NormalizeQuery]
+               [--AllowCompletions]
+               [--Offset]
+               [--Count]
+               [--InterpretationEntityAttributes]
+               [--InterpretationEntityCount]
+               [--Timeout]
+```
+
+### Interpret Required Parameters
+
+`--Query`
+
+The natural lanugage query string to interpret.
+
+`--IndexFilePaths`
+
+The file path expression for specifying which index file(s) to use. Use wild card to specify multiple indexes e.g. './index.*.kes'.
+
+`--GrammarFilePath`
+
+The compiled grammar binary file path. Interpret requires a compiled grammar binary
+
+### Interpret Optional Parameters
+
+`--NormalizeQuery`
+
+Whether normalization rules should be applied to the query before making interpretations.
+
+`--AllowCompletions`
+
+Whether to generate interpretations assuming the query is a partial query that's not yet fully formulated. When set to true, better interpretations can be generated for scenarios such as auto-suggest. 
+
+`--Offset`
+
+The number of top interpretations to be skipped/excluded in the result set.
+
+`--Count`
+
+The number of top interpretations to be included in the result set.
+
+`--SelectInterpretationEntities`
+
+A list of entity attributes, seperated by ','. Use '*' for all entity attributes. Each interpretation in the result set can include the top matching entities used for generating the interpretation. SelectInterpretationEntities specifies which attributes of the top matching entities should be included in the result set.
+
+`--TakeInterpretionEntities`
+
+The number of top matching entities to be included for each interpretation. Each interpretation in the result set can include the top matching entities used for generating the interpretation. TakeInterpretionEntities specifies how many top matching entnties should be included in the result set
+
+`--Timeout`
+
+Maximum amount of time in milliseconds allowed for command to complete before aborting the command. The interpret command will return the top interptations found in the allowed timedout.
+
+## Evaluate Command
+
+Evaluates a KES query expression and returns the top matching entities in the index(es).
+
+```cmd
+kesm Evaluate --KesQueryExpression
+              --IndexFilePaths
+              [--Attributes]
+              [--Skip]
+              [--Take]
+              [--OrderBy]
+              [--OrderByDescending]
+              [--Timeout]
+```
+
+### Evaluate Required Parameters
+
+`--KesQueryExpression`
+
+The KES query expression that specifies entities in the index(es).
```

Comment:
The [query expression](concepts-query-expressions.md) to use for selecting entities from the index(es).